matplotcheck/tests/test_notebook.py

```diff
@@ -0,0 +1,145 @@
+    """Test convert single axes."""
+    fig, ax = plt.subplots()
+    basic_polygon_gdf.plot(ax=ax)
+    store = nb.convert_axes(plt)
+    with pytest.raises(AssertionError, match="Axis lines are displayed on"):
+        PlotTester(store).assert_axis_off()
+    plt.close()
+
+
+def test_notebook_convert_multi_axes(basic_polygon_gdf):
+    """Test convert multi axes."""
+    fig, (ax1, ax2) = plt.subplots(1, 2)
+    basic_polygon_gdf.plot(ax=ax1)
+    basic_polygon_gdf.plot(ax=ax2)
+    store = nb.convert_axes(plt, which_axes="all")
+    for axes in store:
+        with pytest.raises(
+            AssertionError, match="Axis lines are displayed on"
+        ):
+            PlotTester(axes).assert_axis_off()
+    plt.close()
+
+
+def test_notebook_convert_last_axes(basic_polygon_gdf):
+    """Test convert last axes."""
+    fig, (ax1, ax2) = plt.subplots(1, 2)
+    basic_polygon_gdf.plot(ax=ax2)
+    store = nb.convert_axes(plt, which_axes="last")
+    with pytest.raises(AssertionError, match="Axis lines are displayed on"):
+        PlotTester(store).assert_axis_off()
+    plt.close()
+
+
+def test_notebook_convert_first_axes(basic_polygon_gdf):
+    """Test convert first axes."""
+    fig, (ax1, ax2) = plt.subplots(1, 2)
+    basic_polygon_gdf.plot(ax=ax1)
+    store = nb.convert_axes(plt, which_axes="first")
+    with pytest.raises(AssertionError, match="Axis lines are displayed on"):
+        PlotTester(store).assert_axis_off()
+    plt.close()
+
+
+def test_notebook_convert_axes_error(basic_polygon_gdf):
+    """Test convert axes error."""
+    fig, ax = plt.subplots()
+    basic_polygon_gdf.plot(ax=ax)
+    with pytest.raises(ValueError, match="which_axes must be one of the "):
+        nb.convert_axes(plt, which_axes="error")
+    plt.close()
+
+
+def test_error_test_count_pass(capsys):
+    """Test error test prints out correct statement when passing"""
+    nb.error_test(1, 1)
+    captured = capsys.readouterr()
+    assert captured.out == "ERRORS TEST: PASSED!\n"
+
+
+def test_error_test_count(capsys):
+    """Test error test prints out correct statement when failing"""
+    nb.error_test(1, 2)
+    captured = capsys.readouterr()
+    assert (
+        captured.out
+        == "ERRORS TEST: FAILED! 1 of 2 Cells ran without errors\n"
+    )
+
+
+def test_remove_comments_with_comments():
+    """Test remove comments returns string without comments"""
+    test_string = """Hello\n# comments\ntest\nstring"""
+    commentless = nb.remove_comments(test_string)
+    assert commentless == """Hello\ntest\nstring"""
+
+
+def test_remove_comments_without_comments():
+    """Test remove comments leaves commentless string alone"""
+    test_string = """Hello\ntest\nstring"""
+    commentless = nb.remove_comments(test_string)
+    assert commentless == """Hello\ntest\nstring"""
+
+
+def test_import_test_pass(capsys, locals_dictionary_good):
+    """Test import test passes when import are done correctly"""
+    nb.import_test(locals_dictionary_good, 3)
```

Comment:
same question here - what is import_test for?

Reply:
ok i spent some time looking at the code. there are some very cool helper functions in there that we need to document. let's work together to figure out how they are used.  it looks like import_test -- ensure that no imports exist beyond the top cell.

Can you please test this out? and note that sometimes we have markdown at the top of the notebook so let's see that it still passes with markdown cells above the imports cell!! this is cool